doc/decisions/map_preloading.md

```diff
@@ -0,0 +1,51 @@
+# Map Preloading
+
+## Problem
+
+The application's main component is the map, which consists of multiple layers and even more objects. However, not all layers are visible simultaneously, nor are they required during the initial loading of the application. Loading all the data at once during the initial load may result in slower performance, therefore a strategy must be developed to preload data on the map to optimize its performance.
```

Comment:
Please one sentence per line (see documentation guidelines) to make reviews easier:

```suggestion
The application's main component is the map, which consists of multiple layers and even more objects.
Not all layers, however, are visible simultaneously, nor are they required during the initial loading of the application.
Loading all the data at once during the initial load may result in slower performance, therefore a strategy must be developed to preload data on the map to optimize its performance.
```